### DISCUSSION TITLE
request help: can i use resty.http in body_filter send http 1.1 request and parse response body

### DISCUSSION THREAD

**Discussion by zhangweidongyyl** [apache/apisix#5704] [Q&A]
### Issue description

can i use resty.http in body_filter like this
local http = require "resty.http"                               -- 加载 lua-resty-http 库
httpc, err = http:new()                                         -- 创建 http 连接对象
httpc:set_timeout(time)                                         -- 注意名字里有一个下划线
httpc:set_timeout(connect_timeout,                              -- 同上
        send_timeout, read_timeout)

### Environment

- apisix version (cmd: `apisix version`):
- OS (cmd: `uname -a`):
- OpenResty / Nginx version (cmd: `nginx -V` or `openresty -V`):
- etcd version, if have (cmd: run `curl http://127.0.0.1:9090/v1/server_info` to get the info from server-info API):
- apisix-dashboard version, if have:
- the plugin runner version, if the issue is about a plugin runner (cmd: depended on the kind of runner):
- luarocks version, if the issue is about installation (cmd: `luarocks --version`):

**Comment by showcup** [comment]
Yes, you can do that

**Reply by tokers** [reply]
No you can't. This is a limitation of OpenResty. resty.http uses cosocket, which is not available in the body filter phase.
